x/vesting/keeper/integration_test.go

```diff
@@ -55,75 +62,113 @@ var _ = Describe("Clawback Vesting Accounts", Ordered, func() {
+		testAccounts[i] = TestClawbackAccount{
+			privKey: &ethsecp256k1.PrivKey{
+				Key: privKey.Bytes(),
+			},
+			address: address.Bytes(),
+		}
+	}
+
 	var (
 		clawbackAccount *types.ClawbackVestingAccount
 		unvested        sdk.Coins
 		vested          sdk.Coins
 	)
 
+	dest := sdk.AccAddress(tests.GenerateAddress().Bytes())
+	funder := sdk.AccAddress(tests.GenerateAddress().Bytes())
+
 	BeforeEach(func() {
 		s.SetupTest()
 
-		// Create and fund periodic vesting account
-		vestingStart := s.ctx.BlockTime()
-		baseAccount := authtypes.NewBaseAccountWithAddress(addr)
-		funder := sdk.AccAddress(types.ModuleName)
-		clawbackAccount = types.NewClawbackVestingAccount(
-			baseAccount,
-			funder,
-			vestingAmtTotal,
-			vestingStart,
-			lockupPeriods,
-			vestingPeriods,
-		)
-		err := testutil.FundAccount(s.ctx, s.app.BankKeeper, addr, vestingAmtTotal)
-		s.Require().NoError(err)
-		acc := s.app.AccountKeeper.NewAccount(s.ctx, clawbackAccount)
-		s.app.AccountKeeper.SetAccount(s.ctx, acc)
+		gasCoverage := sdk.NewCoins(sdk.NewCoin(stakeDenom, math.NewInt(1e16)))
+
+		// Initialize all test accounts
+		for i, account := range testAccounts {
+			// Create and fund periodic vesting account
+			vestingStart := s.ctx.BlockTime()
+			baseAccount := authtypes.NewBaseAccountWithAddress(account.address)
+			clawbackAccount = types.NewClawbackVestingAccount(
+				baseAccount,
+				funder,
+				vestingAmtTotal,
+				vestingStart,
+				lockupPeriods,
+				vestingPeriods,
+			)
+			testAccounts[i].clawbackAccount = clawbackAccount
 
-		// Check if all tokens are unvested at vestingStart
-		unvested = clawbackAccount.GetUnvestedOnly(s.ctx.BlockTime())
-		vested = clawbackAccount.GetVestedOnly(s.ctx.BlockTime())
-		s.Require().Equal(vestingAmtTotal, unvested)
-		s.Require().True(vested.IsZero())
+			err := testutil.FundAccount(s.ctx, s.app.BankKeeper, account.address, vestingAmtTotal)
+			s.Require().NoError(err)
+			acc := s.app.AccountKeeper.NewAccount(s.ctx, clawbackAccount)
+			s.app.AccountKeeper.SetAccount(s.ctx, acc)
+
+			// Check if all tokens are unvested at vestingStart
+			unvested = clawbackAccount.GetUnvestedOnly(s.ctx.BlockTime())
+			vested = clawbackAccount.GetVestedOnly(s.ctx.BlockTime())
+			s.Require().Equal(vestingAmtTotal, unvested)
+			s.Require().True(vested.IsZero())
+
+			// Grant gas stipend to cover EVM fees
+			err = testutil.FundAccount(s.ctx, s.app.BankKeeper, clawbackAccount.GetAddress(), gasCoverage)
+			s.Require().NoError(err)
+			granteeBalance := s.app.BankKeeper.GetBalance(s.ctx, account.address, stakeDenom)
+			s.Require().Equal(granteeBalance, gasCoverage[0].Add(vestingAmtTotal[0]))
+		}
+
+		// Set clawbackAccount to the first account
+		clawbackAccount = testAccounts[0].clawbackAccount
 	})
 
 	Context("before first vesting period", func() {
 		It("cannot delegate tokens", func() {
-			err := delegate(clawbackAccount, 100)
+			err := delegate(clawbackAccount, math.NewInt(100))
```

Comment:
Each account has extra `spendableBalance` to cover gas fees, which requires an increase in the total vesting quantities, so we can no longer use `int64` to account for values